g2p/apply.py

```diff
@@ -83,12 +129,22 @@ def run(self):
                         "-m",
                         self.g2p_model.get_path(),
                         "-a",
-                        self.word_list.get_path(),
+                        word_list_path,
                     ],
                     stdout=out,
                     stderr=err,
                 )
 
+    def merge(self):
+        g2p_lex_file_list = " ".join(map(tk.uncached_path, self.out_g2p_lexicon_parts))
+        self.sh(f"cat {g2p_lex_file_list} > {tk.uncached_path(self.out_g2p_lexicon)}")
```

Comment:
Please do not use `self.sh`, as you can have arbitrary code execution. You can use `f=open(...)` and then `subprocess.call(["cat"] + g2p_lex_file_list, stdout=f)`

Reply:
>  You can use `f=open(...)` and then

better even:
```
with open(...) as f:
  subprocess.call(["cat"] + g2p_lex_file_list, stdout=f)
```

Reply:
Changed all usages of `self.sh` to `subprocess.check_call` now.